File: 30-days-of-code/cpp/day-20-sorting.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for(int a_i = 0; a_i < n; a_i++){
    	cin >> a[a_i];
    }
    int numberOfSwaps = 0;
    int endPosition = a.size() - 1;
    int swapPosition;
    while( endPosition > 0 ) {
        swapPosition = 0;
        for(int i = 0; i < endPosition; i++) {
            if( a[i] > a[i + 1] ){
                int tmp = a[i];
                a[i] = a[i + 1];
                a[i + 1] = tmp;
                numberOfSwaps += 1;
                swapPosition = i;
            }
        }
        endPosition = swapPosition;
    }
    cout << "Array is sorted in " << numberOfSwaps << " swaps." << endl;
    cout << "First Element: " << a[0] << endl;
    cout << "Last Element: " << a[a.size() - 1] << endl;
    return 0;
}

File: 30-days-of-code/php/day-20-sorting.php
<?php

$handle = fopen ("php://stdin", "r");
fscanf($handle, "%d",$n);
$a_temp = fgets($handle);
$a = explode(" ",$a_temp);
$a = array_map('intval', $a);
$numberOfSwaps = 0;
$endPosition = $n - 1;
$swapPosition;
while( $endPosition > 0 ) {
    $swapPosition = 0;
    for($i = 0; $i < $endPosition; $i++) {
        if( $a[$i] > $a[$i + 1] ){
            $tmp = $a[$i];
            $a[$i] = $a[$i + 1];
            $a[$i + 1] = $tmp;
            $numberOfSwaps += 1;
            $swapPosition = $i;
        }
    }
    $endPosition = $swapPosition;
}
printf("%s%d%s", "Array is sorted in ", $numberOfSwaps, " swaps.\n");
printf("%s%d%s", "First Element: ", $a[0], "\n");
printf("%s%d%s", "Last Element: ", $a[$n- 1], "\n");



?>